### PR TITLE
Feat/navbar progress indicator

### PR DESCRIPTION
**Pull request by Immo-Be**
for #1443 

- introduces chapter navigation indication. Does currently not work for the subchapters (aka giants)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Climate from Space application for ESAs CCI+ program.",
   "main": "dist-src-electron/main.js",
   "scripts": {
-    "start": "vite --port 8080",
+    "start": "vite --port 8080 --host",
```
Comment (andreashelms) [inline package.json:7]: What is that needed for?

File: src/scripts/components/stories/pilot-story/components/chapter-progress-indication/chapter-progress-indication.tsx [hunk 2]
```diff
@@ -1,29 +1,81 @@
...
+
+import {useChapter} from '../../hooks/use-chapter';
+
+import {dataIsTitleInView, progressIndicationElement} from '../../config/main';
+import {ChapterType} from '../../types/globe';
 
 import cx from 'classnames';
-import styles from './chapter-progress-indication.module.styl';
 
+import styles from './chapter-progress-indication.module.styl';
```
Comment (andreashelms) [inline src/scripts/components/stories/pilot-story/components/chapter-progress-indication/chapter-progress-indication.tsx:10]:
```suggestion
import styles from './chapter-progress-indication.module.styl';

```

File: src/scripts/components/stories/pilot-story/components/nav-drawer/nav-drawer.tsx [hunk 3]
```diff
@@ -16,10 +10,11 @@ import {getSnapPoint} from '../utils/nav-drawer';
...
 import styles from './nav-drawer.module.styl';
-
-interface Props extends PropsWithChildren {
-  handle: React.JSX.Element;
-}
+import {GlobeIcon} from '../../../../main/icons/globe-icon';
+import {chapters} from '../../config/main';
+import Button from '../button/button';
+import {GlobeExploreIcon} from '../icons/globe-explore-icon';
+import NavChapterOverview from '../nav-chapter-overview/nav-chapter-overview';
```
Comment (andreashelms) [inline src/scripts/components/stories/pilot-story/components/nav-drawer/nav-drawer.tsx:17]: Please sort imports

File: src/scripts/components/stories/pilot-story/provider/chapter-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  createContext,
+  useEffect,
+  useState
+} from 'react';
+import {useHistory} from 'react-router-dom';
+import {useParallaxController} from 'react-scroll-parallax';
+import useIntersectChapters from '../hooks/use-chapter-observer';
+import {chapterMainElement} from '../config/main';
+
+function useChapterContext() {
```
Comment (andreashelms) [inline src/scripts/components/stories/pilot-story/provider/chapter-provider.tsx:13]: Nice! I think I will refactor globe component to use this hook for obtaining chapter progress.

Comment (Immo-Be) [reply src/scripts/components/stories/pilot-story/provider/chapter-provider.tsx:13]: cool!

File: package.json (hunk 1 above)
Comment (Immo-Be) [reply package.json]: FYI: I removed the flag

**Review by andreashelms — APPROVED** [review]

**Pull request merged**